src/components/TextField/TextArea.tsx

```diff
@@ -1,45 +1,66 @@
 import React from "react";
-import { TextArea as BlueprintTextArea, Intent as BlueprintIntent } from "@blueprintjs/core";
+import {
+  TextArea as BlueprintTextArea,
+  Intent as BlueprintIntent,
+  TextAreaProps as BlueprintTextAreaProps,
+} from "@blueprintjs/core";
 import { CLASSPREFIX as eccgui } from "../../configuration/constants";
 
+interface TextAreaProps extends Partial<BlueprintTextAreaProps> {
+  /**
+   * when set to true the input takes a blue border color
+   */
+  hasStatePrimary?: boolean;
+  /**
+   * when set to true the input takes a green border color
+   */
+  hasStateSuccess?: boolean;
+  /**
+   * when set to true the input takes an orange border color
+   */
+  hasStateWarning?: boolean;
+  /**
+   * when set to true the input takes a red border color
+   */
+  hasStateDanger?: boolean;
+}
+
 function TextArea({
-    className = "",
-    hasStatePrimary = false,
-    hasStateSuccess = false,
-    hasStateWarning = false,
-    hasStateDanger = false,
-    fullWidth = false,
-    rows = 5,
-    ...otherProps
-}: any) {
-    let intent;
-    switch (true) {
-        case hasStatePrimary:
-            intent = BlueprintIntent.PRIMARY;
-            break;
-        case hasStateSuccess:
-            intent = BlueprintIntent.SUCCESS;
-            break;
-        case hasStateWarning:
-            intent = BlueprintIntent.WARNING;
-            break;
-        case hasStateDanger:
-            intent = BlueprintIntent.DANGER;
-            break;
-        default:
-            break;
-    }
+  className = "",
+  hasStatePrimary = false,
+  hasStateSuccess = false,
+  hasStateWarning = false,
+  hasStateDanger = false,
+  rows = 5,
+  ...otherProps
+}: TextAreaProps) {
+  let intent;
+  switch (true) {
```

Comment:
This code is older, not added in this feature branch. Yes, as like all switch statements it could be replaced by a chain of if-then-else statements but at the end the result would be the same.